[TEP074] Remove Image pipelineResources (#6002)
tektoncd/pipeline: docs/resources.md

### Image Resource

> Image Resource is deprecated and please refer to the doc at https://github.com/tektoncd/pipeline/blob/main/docs/pipelineresources.md#replacing-an-image-resource for migration.

Reviewer: ```suggestion
Support for image resources has been removed. Please refer to the doc at https://github.com/tektoncd/pipeline/blob/main/docs/pipelineresources.md#replacing-an-image-resource for migration.
```

suggesting since "deprecation" implies that it still works, while "removal" implies that it does not.
